### PR TITLE
feat: add Fact, add sending longevity facts to server, rename Stats to CommitStats, rename Commits to CommitsGroup (APP-74)

### PR DESCRIPTION
**Pull request by anatolykislov**

### CODE REVIEW

File: src/main/kotlin/app/hashers/CodeLongevity.kt [hunk 1]
```diff
@@ -98,24 +103,51 @@ class CodeLongevity(private val localRepo: LocalRepo,
...
+            sums[email] = sums[email]!! + line.age
+            totals[email] = totals[email]!! + 1
         }
 
+        val repoAvg = if (repoTotal > 0) { repoSum / repoTotal } else 0
+        val stats = mutableListOf<Fact>()
+        stats.add(Fact(repo = serverRepo,
+                       key = CODE_LONGEVITY_REPO_KEY,
+                       value = repoAvg.toDouble()))
+        println("Repo average code line age is $repoAvg seconds, "
```
Comment (sergey-surkov) [inline src/main/kotlin/app/hashers/CodeLongevity.kt]: Shouldn't this be log? And why seconds? It should be days at least :)

File: src/main/kotlin/app/hashers/CodeLongevity.kt [hunk 2]
```diff
@@ -98,24 +103,51 @@ class CodeLongevity(private val localRepo: LocalRepo,
...
-            println("Average code line age for <$email> is $avg seconds, "
-                  + "lines total: $total")
+            val total = totals[email] ?: 0
+            val avg = if (total > 0) { sums[email]!! / total } else 0
+            stats.add(Fact(repo = serverRepo,
+                           key = CODE_LONGEVITY_KEY,
+                           value = avg.toDouble(),
+                           author = Author(email = email)))
+            if (email == localRepo.author.email) {
+                println("Your average code line age is $avg seconds, "
```
Comment (sergey-surkov) [inline src/main/kotlin/app/hashers/CodeLongevity.kt]: log?

File: src/main/kotlin/app/model/Fact.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.security.InvalidParameterException
+
+/**
+ * Key-value statistics for fun facts.
+ */
+data class Fact(
+    var repo: Repo = Repo(),
+    var key: String = "",
+    var value: Double = 0.0,
+    var author: Author = Author()
```
Comment (sergey-surkov) [inline src/main/kotlin/app/model/Fact.kt]: I do think this should have an optional expiration date.

File: src/main/kotlin/app/hashers/CodeLongevity.kt (hunk 1 above)
Comment (anatolykislov) [reply src/main/kotlin/app/hashers/CodeLongevity.kt]: Yep, you're right.

File: src/main/kotlin/app/hashers/CodeLongevity.kt (hunk 2 above)
Comment (anatolykislov) [reply src/main/kotlin/app/hashers/CodeLongevity.kt]: Same.

File: src/main/kotlin/app/model/Fact.kt (hunk 3 above)
Comment (anatolykislov) [reply src/main/kotlin/app/model/Fact.kt]: Better to use DB timestamps for expiration calculation.

Comment (sergey-surkov) [reply src/main/kotlin/app/model/Fact.kt]: I disagree. Timestamp could depend on what a fact is. And it's better to keep fact-specific knowledge to as few places as possible. The app knows best when a fact should expire.

Comment (anatolykislov) [reply src/main/kotlin/app/model/Fact.kt]: Could you give an example for facts with different expiration interval?

Comment (sergey-surkov) [reply src/main/kotlin/app/model/Fact.kt]: Some facts can be transient. E.g. "I just used C++11 for the first time". This should expire in a month or so.

**Comment by sergey-surkov** [comment]
BTW, feel free to merge it. Expiration date on fun facts can always be added later. It's not a showstopper, that's why I approved it yesterday.

File: src/main/kotlin/app/model/Fact.kt (hunk 3 above)
Comment (anatolykislov) [reply src/main/kotlin/app/model/Fact.kt]: Good example. Let's discuss it offline.

**Pull request merged**